File: test_mark3labs.go
package main

import (
	"context"
	"fmt"
	"log"
	"os"
	"os/exec"
	"strings"
	"time"

	"github.com/mark3labs/mcp-go/client"
	"github.com/mark3labs/mcp-go/client/transport"
	"github.com/mark3labs/mcp-go/mcp"
)

func main() {
	fmt.Println("🧪 Testing Mark3Labs Weather MCP Server")
	fmt.Println(strings.Repeat("=", 50))

	// Start the Mark3Labs server as a subprocess
	cmd := exec.Command("go", "run", "main.go")
	cmd.Dir = "/Users/hulkster/WeatherFull/mark3labs-implementation"
	cmd.Env = append(os.Environ(), "WEATHER_API_KEY=f5fc1e71ee4a4963a74195621252607")

	// Get stdin/stdout pipes
	stdin, err := cmd.StdinPipe()
	if err != nil {
		log.Fatalf("Failed to get stdin pipe: %v", err)
	}

	stdout, err := cmd.StdoutPipe()
	if err != nil {
		log.Fatalf("Failed to get stdout pipe: %v", err)
	}

	// Start the server
	if err := cmd.Start(); err != nil {
		log.Fatalf("Failed to start server: %v", err)
	}
	defer cmd.Process.Kill()

	// Create STDIO transport
	stdioTransport := transport.NewStdio(stdin, stdout, nil, "mark3labs-weather-server")

	// Create a new MCP client
	c := client.NewClient(stdioTransport)

	// Start the client with timeout
	ctx, cancel := context.WithTimeout(context.Background(), 15*time.Second)
	defer cancel()

	if err := c.Start(ctx); err != nil {
		log.Fatalf("Failed to start client: %v", err)
	}
	defer c.Close()

	// Initialize the client
	initRequest := mcp.InitializeRequest{
		Params: mcp.InitializeParams{
			ProtocolVersion: "2024-11-05",
			Capabilities: mcp.ClientCapabilities{
				Experimental: map[string]any{
					"tools": true,
				},
			},
			ClientInfo: mcp.Implementation{
				Name:    "mark3labs-test-client",
				Version: "1.0.0",
			},
		},
	}

	_, err = c.Initialize(ctx, initRequest)
	if err != nil {
		log.Fatalf("Failed to initialize client: %v", err)
	}

	// List available tools
	toolsRequest := mcp.ListToolsRequest{}
	toolsResult, err := c.ListTools(ctx, toolsRequest)
	if err != nil {
		log.Fatalf("Failed to list tools: %v", err)
	}

	fmt.Println("✅ Mark3Labs Server Test Results:")
	fmt.Printf("Available tools: %d\n", len(toolsResult.Tools))
	for _, tool := range toolsResult.Tools {
		fmt.Printf("- %s: %s\n", tool.Name, tool.Description)
	}

	// Test the get_current_weather tool
	if len(toolsResult.Tools) > 0 {
		fmt.Println("\n🧪 Testing get_current_weather tool...")

		request := mcp.CallToolRequest{
			Params: mcp.CallToolParams{
				Name: "get_current_weather",
				Arguments: map[string]interface{}{
					"city": "Tokyo",
				},
			},
		}

		result, err := c.CallTool(ctx, request)
		if err != nil {
			log.Printf("❌ Failed to call tool: %v", err)
		} else {
			fmt.Println("✅ Tool call successful!")
			if len(result.Content) > 0 {
				if textContent, ok := result.Content[0].(mcp.TextContent); ok {
					fmt.Printf("📄 Response: %s\n", textContent.Text)
				}
			}
		}
	}

	fmt.Println("\n🎉 Mark3Labs server test completed successfully!")
	fmt.Println("✅ mark3labs/mcp-go SDK implementation working!")
	os.Exit(0)
}

File: test_mark3labs.py
#!/usr/bin/env python3
import json
import subprocess
import sys

def test_mark3labs_weather(city):
    """Test the Mark3Labs weather MCP server"""
    
    # Set up the MCP request
    init_request = {
        "jsonrpc": "2.0",
        "id": 1,
        "method": "initialize",
        "params": {
            "protocolVersion": "2024-11-05",
            "capabilities": {},
            "clientInfo": {"name": "test", "version": "1.0"}
        }
    }
    
    weather_request = {
        "jsonrpc": "2.0",
        "id": 2,
        "method": "tools/call",
        "params": {
            "name": "get_current_weather",
            "arguments": {"city": city}
        }
    }
    
    try:
        # Set environment variable
        env = {"WEATHER_API_KEY": "f5fc1e71ee4a4963a74195621252607"}
        
        # Start the server process
        process = subprocess.Popen(
            ["./mark3labs-weather-server"],
            stdin=subprocess.PIPE,
            stdout=subprocess.PIPE,
            stderr=subprocess.PIPE,
            env=env,
            cwd="mark3labs-implementation"
        )
        
        # Send initialization
        process.stdin.write((json.dumps(init_request) + "\n").encode())
        process.stdin.flush()
        
        # Read response
        init_response = process.stdout.readline().decode().strip()
        print(f"Init response: {init_response}")
        
        # Send weather request
        process.stdin.write((json.dumps(weather_request) + "\n").encode())
        process.stdin.flush()
        
        # Read response
        weather_response = process.stdout.readline().decode().strip()
        print(f"Weather response: {weather_response}")
        
        # Parse and extract weather data
        response_data = json.loads(weather_response)
        if "result" in response_data and "content" in response_data["result"]:
            weather_text = response_data["result"]["content"][0]["text"]
            return weather_text
        
        process.terminate()
        return None
        
    except Exception as e:
        print(f"Error getting weather for {city}: {e}")
        return None

def main():
    """Test weather for multiple cities"""
    
    cities = ["Tokyo", "London", "New York"]
    
    print("🌤️  Mark3Labs Weather MCP Server Test")
    print("=" * 50)
    
    for city in cities:
        print(f"\n📍 Testing: {city}")
        weather = test_mark3labs_weather(city)
        if weather:
            print(f"✅ Success!")
            print(weather)
        else:
            print(f"❌ Failed to get weather for {city}")
    
    print("\n" + "=" * 50)
    print("✅ Mark3Labs Weather MCP Server is working!")
    print("🌍 Supports international locations")

if __name__ == "__main__":
    main() 